### PR TITLE
[DOCS] puts 1.0 code under test for how to define Expectations

### PR DESCRIPTION
**Pull request by Rachel-Reverie**
## Description
- Puts 1.0 code into scripts under test for:
   - How to create an Expectation with preset parameters
   - How to create an Expectation that uses Expectation Parameters
   - How to retrieve a Batch of sample data for testing Expectations using the `pandas_default` Data Source
   - How to retrieve a Batch of sample data for testing Expectations from a Batch Definition
   - How to test and modify an Expectation that uses preset parameters
   - How to test and modify an Expectation that uses an Expectation Parameters dictionary
   - How to organize Expectations into Expectation Suites

## Definition of done:
- [X] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [X] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [X] Appropriate tests and docs have been updated

### CODE REVIEW

File: docs/docusaurus/docs/core/define_expectations/_examples/create_an_expectation.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+# </snippet>
+
+# In this case, two Expectations are created that will be passed
+#  parameters at runtime, and unique lookups are defined for each
+#  Expectations' parameters.
+
+# <snippet name="docs/docusaurus/docs/core/define_expectations/_examples/create_an_expectation.py - dynamic expectations">
+passenger_expectation = gx.expectations.ExpectColumnMaxToBeBetween(
+    column="passenger_count",
+    min_value={"$PARAMETER": "expect_passenger_max_to_be_above"},
```
Comment (tyler-hoffman) [inline docs/docusaurus/docs/core/define_expectations/_examples/create_an_expectation.py:36]: Just floating an idea: Does this become easier if you make a constant for `expect_passenger_max_to_be_above`, etc, to show how they are linked to what is passed in below?

File: docs/docusaurus/docs/core/define_expectations/_examples/create_an_expectation.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    max_value={"$PARAMETER": "expect_passenger_max_to_be_below"},
+)
+fare_expectation = gx.expectations.ExpectColumnMaxToBeBetween(
+    column="fare",
+    min_value={"$PARAMETER": "expect_fare_max_to_be_above"},
```
Comment (tyler-hoffman) [inline docs/docusaurus/docs/core/define_expectations/_examples/create_an_expectation.py:41]: Looking at this again, I'm wondering if we should consider abstracting away the `$PARAMETER` part of this before launch? Like what if it just became `min_value=gx.expectations.parameter("expect_fare_max_to_be_above"),`? Not blocking for this PR at all.

File: docs/docusaurus/docs/core/define_expectations/test_an_expectation.md [hunk 3]
```diff
@@ -131,6 +130,11 @@ Data can be validated against individual Expectations.  This workflow is general
    }
    ```
 
+<!--    When an Expectation uses an Expectation Parameter dictionary you don't have to modify anything on the Expectation object.  Instead, update the dictionary with new values and then test it with the updated dictionary:
```
Comment (tyler-hoffman) [inline docs/docusaurus/docs/core/define_expectations/test_an_expectation.md:133]: Is this commented out, or is this syntax something else in markdown?

**Review by tyler-hoffman — COMMENTED** [review]
Left a couple comments. Nothing blocking, but figured I'd verify if the one part is commented out before approving

File: docs/docusaurus/docs/core/define_expectations/test_an_expectation.md (hunk 3 above)
Comment (Rachel-Reverie) [reply docs/docusaurus/docs/core/define_expectations/test_an_expectation.md:133]: It's commented out.  I confirmed with Bill earlier today that we can't currently test individual Expectation Parameter using Expectations like this, but there is a ticket for implementing that post 1.0, so I decided to leave the content there but not rendered for the site.  When the implementation is in place we can just un-comment this.

File: docs/docusaurus/docs/core/define_expectations/_examples/create_an_expectation.py (hunk 1 above)
Comment (Rachel-Reverie) [reply docs/docusaurus/docs/core/define_expectations/_examples/create_an_expectation.py:36]: As in:
```
EXPECT_PASSENGER_MAX_TO_BE_ABOVE="expect_passenger_max_to_be_above"

...

passenger_expectation = gx.expectations.ExpectColumnMaxToBeBetween(
    column="passenger_count",
    min_value={"$PARAMETER": EXPECT_PASSENGER_MAX_TO_BE_ABOVE},

...

expectation_parameters = {
    EXPECT_PASSENGER_MAX_TO_BE_ABOVE: 1
}
```
?

I'm not certain if that's easier to comprehend or not, but I'm not against making the change if you think it is.

File: docs/docusaurus/docs/core/define_expectations/test_an_expectation.md (hunk 3 above)
Comment (Rachel-Reverie) [reply docs/docusaurus/docs/core/define_expectations/test_an_expectation.md:133]: I'm not opposed to removing it entirely, if you think that would be preferable.  If I do I'd probably look up the ticket for adding that functionality and copy the snippets into a comment on it.

Comment (tyler-hoffman) [reply docs/docusaurus/docs/core/define_expectations/test_an_expectation.md:133]: Totally your call!

File: docs/docusaurus/docs/core/define_expectations/_examples/create_an_expectation.py (hunk 1 above)
Comment (tyler-hoffman) [reply docs/docusaurus/docs/core/define_expectations/_examples/create_an_expectation.py:36]: Maybe something like `MAX_PASSENGER_PARAMETER`, but that's the idea. I don't have a strong opinion here, I guess. If this were production code, I'd do it, but maybe that doesn't translate perfectly to documentation. Totally your call.

**Review by tyler-hoffman — APPROVED** [review]

**Pull request merged**